index.js

```diff
@@ -38,153 +39,178 @@ module.exports.s3 = function(config, key, args, callback) {
 	);
 };
 
+var getDimArray = function( dims ) {
+	var dimArr = ( typeof dims === 'string'
+		? dims.split(',')
+		: dims
+	);
+	return dimArr.map(function(v) {
+		return Number(v) || null;
```

Comment:
Also, can this just `.map( Number )` instead? What's the implication of `NaN` vs `null`?

Reply:
Yep, `null` or `undefined` to use an automatic value with sharp. http://sharp.pixelplumbing.com/en/stable/api-resize/#parameters